Add contact form

jia1 opened the pull request:
- [x] Add contact form to Connect page
- [ ] List down other ways to contact us on the Connect page (i.e. address, hotline, defer to next PR)
- [x] Create function for submit
- [ ] Google maps (defer decision to next PR)

Signed-off-by: ♫ JIA YEE ♚ <jiayeerawr@gmail.com>

jia1 committed:
Add contact form (frontend only)

TODO:
1. Submit button onclick
2. List down other ways to contact us
3. Google maps

Signed-off-by: ♫ JIA YEE ♚ <jiayeerawr@gmail.com>

jia1 committed:
Add AJAX call to handle contact form submission

Signed-off-by: ♫ JIA YEE ♚ <jiayeerawr@gmail.com>

Skaty commented on layouts/partials/form.html:
```diff
@@ -69,4 +69,25 @@
     $('select').material_select();
   });
+
+  function connect() {
```
Why not use 
```javascript
$(<form selector>).serialize()
```
[Relevant jQuery docs](https://api.jquery.com/serialize/)

jia1 replied:
So you mean instead of extracting the fields one by one, I just serialize the form input values? Okay! Thank you @Skaty

Skaty reviewed (commented):
@jia1

jia1 committed:
Add subject field, form ID, and AJAX url

Signed-off-by: ♫ JIA YEE ♚ <jiayeerawr@gmail.com>

jia1 committed:
Replace manual selection of input val to serialize()

Signed-off-by: ♫ JIA YEE ♚ <jiayeerawr@gmail.com>

jia1 committed:
Fix serialize output

serialize returns URL query string, but we want a JSON string
so we do serializeArray and process the array into the JSON object
and then stringify it

Signed-off-by: ♫ JIA YEE ♚ <jiayeerawr@gmail.com>

jia1 commented:
@leeyimin we need your review! Thank you
